core/src/main/scala/org/broadinstitute/dsde/rawls/billing/GoogleBillingProjectCreator.scala

```diff
@@ -0,0 +1,42 @@
+package org.broadinstitute.dsde.rawls.billing
+
+import akka.http.scaladsl.model.StatusCodes
+import org.broadinstitute.dsde.rawls.dataaccess.{GoogleServicesDAO, SamDAO}
+import org.broadinstitute.dsde.rawls.model.{CreateRawlsV2BillingProjectFullRequest, ErrorReport, ErrorReportSource, SamResourceTypeNames, SamServicePerimeterActions, ServicePerimeterName, UserInfo}
+import org.broadinstitute.dsde.rawls.user.UserService.syncBillingProjectOwnerPolicyToGoogleAndGetEmail
+
+import java.net.URLEncoder
+import scala.concurrent.{ExecutionContext, Future}
+import java.nio.charset.StandardCharsets.UTF_8
+
+class GoogleBillingProjectCreator(samDAO: SamDAO, gcsDAO: GoogleServicesDAO)(implicit executionContext: ExecutionContext) extends BillingProjectCreator {
+  implicit val errorReportSource: ErrorReportSource = ErrorReportSource("rawls")
+
+  override def validateBillingProjectCreationRequest(createProjectRequest: CreateRawlsV2BillingProjectFullRequest, userInfo: UserInfo): Future[Unit] = {
+    for {
+      _ <- GoogleBillingProjectCreator.checkServicePerimeterAccess(createProjectRequest.servicePerimeter, samDAO, userInfo)
+      hasAccess <- gcsDAO.testBillingAccountAccess(createProjectRequest.billingAccount.get, userInfo)
+      _ = if (!hasAccess) {
+        throw new GoogleBillingAccountAccessException(ErrorReport(StatusCodes.BadRequest, "Billing account does not exist, user does not have access, or Terra does not have access"))
+      }
+    } yield {}
+  }
+
+
+  override def postCreationSteps(createProjectRequest: CreateRawlsV2BillingProjectFullRequest, userInfo: UserInfo): Future[Unit] = {
+    for {
+      _ <- syncBillingProjectOwnerPolicyToGoogleAndGetEmail(samDAO, createProjectRequest.projectName)
+    } yield {}
+  }
+}
+
+object GoogleBillingProjectCreator {
+  def checkServicePerimeterAccess(servicePerimeterOption: Option[ServicePerimeterName], samDAO: SamDAO, userInfo: UserInfo)(implicit executionContext: ExecutionContext): Future[Unit] = {
+    servicePerimeterOption.map { servicePerimeter =>
```

Comment:
This used to call `ServicePerimeterService.checkServicePerimeterAccess(samDAO, createProjectRequest.servicePerimeter, userInfo)`. Why did you decide to implement something custom here?

Reply:
Ugh, that's a mistake, I think an artifact from a rebase from a related branch. I will remove this dupe method.